creator-node/src/ipfsClient.js

```diff
@@ -27,69 +25,121 @@ async function logIpfsPeerIds () {
- * @param {Buffer|Object[]} inputData either a single buffer, or an Object[] with the structure { path: string, content: buffer }
+ * @param {Buffer} inputData a single buffer input
  * @param {Object?} ipfsConfig ipfs add config options
  * @param {Object?} logContext
- * @param {boolean?} addToIPFSDaemon flag to override adding to ipfs daemon
- * @returns {string|string[]|Object|Object[]} hashes from content addressing fn, or ipfs daemon responses
+ * @param {boolean?} enableIPFSAdd flag to add content to ipfs daemon
+ * @returns {string} hash from content addressing fn, or ipfs daemon response
  */
-async function ipfsAddWrapper (ipfs, inputData, ipfsConfig = {}, logContext = {}, addToIPFSDaemon = false) {
+async function ipfsSingleAddWrapper (ipfs, inputData, ipfsConfig = {}, logContext = {}, enableIPFSAdd = false) {
   const logger = genericLogger.child(logContext)
 
-  // If async ipfs add is enabled, or if `addToIPFSDaemon` flag passed in is false, asynchronously
-  // add to ipfs. Else, synchronously add to ipfs and wait for the response.
-  let ipfsDaemonHashes
-  if (ENABLE_ASYNC_IPFS_ADD && !addToIPFSDaemon) {
-    ipfs.add(inputData, ipfsConfig) // Do not await it
+  // Generate hash with ipfs content hashing logic
+  const onlyHash = await ipfsHashOf(inputData)
+
+  // If async ipfs add is enabled, synchronously add to ipfs.
+  let ipfsDaemonHash
+  if (enableIPFSAdd) {
+    try {
+      ipfsDaemonHash = (await ipfs.add(inputData, ipfsConfig))[0].hash
+    } catch (e) {
+      logger.warn(`[ipfsClient - ipfsSingleAddWrapper()] Could not add content to ipfs. Defaulting to onlyHash=${onlyHash}: ${e.toString()}`)
+      return onlyHash
+    }
+  }
+
+  // Return the `ipfsDaemonHashes`, or `onlyHashes`. Prioritize `ipfsDaemonHashes`.
+  if (ipfsDaemonHash) {
+    logger.info(`[ipfsClient - ipfsSingleAddWrapper()] onlyHash=${onlyHash} ipfsDaemonHash=${ipfsDaemonHash} isSameHash=${onlyHash === ipfsDaemonHash}`)
+    return ipfsDaemonHash
   } else {
-    ipfsDaemonHashes = await ipfs.add(inputData, ipfsConfig)
+    logger.info(`[ipfsClient - ipfsSingleAddWrapper()] onlyHash=${onlyHash}`)
+    return onlyHash
   }
+}
+
+// TODO: Replace for images and figure out correct response structure
+/**
+ * Wrapper to ipfs.add() -- Allows enabling/disabling adding content to the ipfs daemon. Generally used for images (to generate dirs too)
+ * @param {Object} ipfs ipfs library -- can be two different versions (see ipfsClient.js)
+ * @param {Object[]} inputData an Object[] with the structure { path: string, content: buffer }
+ * @param {Object?} ipfsConfig ipfs add config options
+ * @param {Object?} logContext
+ * @param {boolean?} enableIPFSAdd flag to add content to ipfs daemon
+ * @returns {string|string[]|Object|Object[]} hashes from content addressing fn, or ipfs daemon responses
+ */
+// eslint-disable-next-line no-unused-vars
+async function ipfsMultipleAddWrapper (ipfs, inputData, ipfsConfig = {}, logContext = {}, enableIPFSAdd = false) {
```

Comment:
tentative usage for adding to images

Reply:
not used anywhere yet

Reply:
lets def get this right and get it consumed, which part of this are you unsure about?

Reply:
understood, i wasn't aware of a v3. per above comment, would be in favor of doing in v2 but sounds like you all have discussed, good either way 👍